diff -Nur 1/src/package.cpp 2/src/package.cpp
--- 1/src/package.cpp	2022-12-26 13:25:44.000000000 +0800
+++ 2/src/package.cpp	2022-12-26 11:12:50.000000000 +0800
@@ -1063,6 +1063,23 @@
    return provides;
 }
 
+QMap<QString, QString> Package::providesListEnhance() const
+{
+    pkgDepCache::StateCache &State = (*d->backend->cache()->depCache())[d->packageIter];
+    if (!State.CandidateVer) {
+        return {};
+    }
+
+    QMap<QString, QString> provides;
+
+    for (pkgCache::PrvIterator Prv =
+         State.CandidateVerIter(*d->backend->cache()->depCache()).ProvidesList(); !Prv.end(); ++Prv) {
+        provides.insert(QLatin1String(Prv.Name()), QLatin1String(Prv.ProvideVersion()));
+    }
+
+   return provides;
+}
+
 QStringList Package::recommendsList() const
 {
     QStringList recommends;
diff -Nur 1/src/package.h 2/src/package.h
--- 1/src/package.h	2022-12-26 13:25:44.000000000 +0800
+++ 2/src/package.h	2022-12-26 11:11:09.000000000 +0800
@@ -475,6 +475,14 @@
     */
     QStringList providesList() const;
 
+    /**
+     * Returns a list of the names of all the virtual packages that this package
+     * provides.
+     *
+     * \return A \c QMap<QString, QString> of packages that this package provides and versions.
+     */
+    QMap<QString, QString> providesListEnhance() const;
+
    /**
     * Returns a list of the names of all the packages that this package recommends.
     *
